openpype/hosts/nuke/api/lib.py

```diff
@@ -3397,3 +3397,24 @@ def create_viewer_profile_string(viewer, display=None, path_like=False):
     if path_like:
         return "{}/{}".format(display, viewer)
     return "{} ({})".format(viewer, display)
+
+
+def get_head_filename_without_hashes(original_path, name):
+    """Function to get the renamed head filename without frame hashes
+    To avoid the system being confused on finding the filename with
+    frame hashes if the head of the filename has the hashed symbol
+
+    Args:
+        original_path (str): the filename with frame hashes
+        name (str): the name of the tags
+
+    Returns:
+        filename: the renamed filename with the tag
+    """
+    filename = os.path.basename(original_path)
+    fhead = filename.split(".")[0]
+    if "#" in fhead:
+        fhead = fhead.replace("#", "")[:-1]
```

Comment:
```suggestion
    if "#" in fhead:
        fhead = fhead.replace("#", "").rstrinp(".")
```
this might be more safe. Since the dot might or might not be there.